CDP4PluginInstaller/App.xaml.cs

```diff
@@ -0,0 +1,225 @@
+//    License as published by the Free Software Foundation; either
+//    version 3 of the License, or any later version.
+//
+//    The CDP4-IME Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//    GNU Affero General Public License for more details.
+//
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+
+namespace CDP4PluginInstaller
+{
+    using System;
+    using System.Collections.Generic;
+    using System.ComponentModel.Composition;
+    using System.ComponentModel.Composition.Hosting;
+    using System.Linq;
+    using System.Windows;
+    using System.IO;
+    using System.IO.Compression;
+    using System.Reflection;
+    using System.Security.Claims;
+    using System.Security.Principal;
+
+    using CDP4PluginInstaller.Utilities;
+    using CDP4PluginInstaller.ViewModels;
+    using CDP4PluginInstaller.Views;
+
+    using CommonServiceLocator;
+
+    using Newtonsoft.Json;
+
+    using NLog;
+
+    /// <summary>
+    /// Interaction logic for App.xaml
+    /// </summary>
+    public partial class App
+    {
+        /// <summary>
+        /// Holds the windows Elevated rights user claim value
+        /// </summary>
+        private const string AdministratorClaimsValue = "S-1-5-32-544";
+
+        /// <summary>
+        /// The NLog logger
+        /// </summary>
+        private Logger logger = LogManager.GetCurrentClassLogger();
+
+        /// <summary>
+        /// Holds the path to the download folder of plugins
+        /// </summary>
+        private readonly string downloadFolder;
+
+        /// <summary>
+        /// Holds the path to the download folder of plugins
+        /// </summary>
+        private Version currentImeVersion;
+
+        /// <summary>
+        /// Holds an <see cref="IEnumerable{T}"/> of type <code>(string pluginDownloadFullPath, Manifest theNewManifest, bool isImeCompatible)</code>
+        /// of the updatable plugins
+        /// </summary>
+        public List<(FileInfo pluginDownloadFullPath, Manifest theNewManifest)> UpdatablePlugins = new List<(FileInfo pluginDownloadFullPath, Manifest theNewManifest)>();
+
+        /// <summary>
+        /// Holds the main Windows instance
+        /// </summary>
+        private MainWindow mainWindow;
+
+        /// <summary>
+        /// Initiate a new <see cref="App"/>
+        /// </summary>
+        public App()
+        {
+            var appData = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+            this.downloadFolder = Path.Combine(appData, "RHEA/CDP4/DownloadCache/Plugins/");
```

Comment:
you are assuming that this is the place where software is installed, is that always the case?